lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1199,17 +1199,16 @@ namespace Js
 
     BOOL JavascriptOperators::GetOwnProperty(Var instance, PropertyId propertyId, Var* value, ScriptContext* requestContext)
     {
-        BOOL result;
-        if (TaggedNumber::Is(instance))
-        {
-            result = false;
-        }
-        else
+        if (!TaggedNumber::Is(instance))
         {
             RecyclableObject* object = RecyclableObject::FromVar(instance);
-            result = object && object->GetProperty(object, propertyId, value, NULL, requestContext);
+            if (object)
+            {
+                return Property_Found == object->GetPropertyEx(object, propertyId, value, NULL, requestContext); // return false if instance is typed arrary and GetPropertyEx() returns Property_NotFound_NoProto
```

Comment:
That is true.
Initially RecyclableObject::GetProperty() wrapper was checking "!= Property_NotFound", and later I determined using "== Property_Found" makes more sense. And the change here used to be necessary.
There are other occurrences of the same scenario. I will change them back. Thanks!

---
In reply to: [105235436](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r105235436) [](ancestors = 105235436)